libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/utils/LocaleUtils.java

```diff
@@ -52,13 +61,24 @@ public static Locale getLocale(Context context) {
   public static void setLocale(Context context, Locale locale, @NavigationUnitType.UnitType int unitType) {
     SharedPreferences.Editor editor = PreferenceManager.getDefaultSharedPreferences(context).edit();
 
+    setLocale(editor, locale, unitType);
+
+    editor.apply();
+  }
+
+  /**
+   * Sets locale and unitType into SharedPreferences for later use
+   * @param editor to add locale and unitType to
+   * @param locale to set language
+   * @param unitType to set unitType to use
+   */
+  public static void setLocale(
+    SharedPreferences.Editor editor, Locale locale, @NavigationUnitType.UnitType int unitType) {
     editor.putInt(NavigationConstants.NAVIGATION_VIEW_UNIT_TYPE, unitType);
```

Comment:
Is there any reason the logic to save the unit type to the preferences is included in this method?  Is it possible to extract the `unitType` into a different method?  `LocaleUtils#setLocale` doesn't give any indication on what is being done with the unit type unless you come to the javadoc / actual method logic